src/r3/atomic-swap/corda/evm-bridge-workflows/src/test/kotlin/com/r3/corda/evmbridge/ProofTests.kt

```diff
@@ -0,0 +1,77 @@
+package com.r3.corda.evmbridge
+
+import com.r3.corda.cno.evmbridge.dto.TransactionReceipt
+import com.r3.corda.evmbridge.internal.TestNetSetup
+import com.r3.corda.evmbridge.workflows.*
+import net.corda.core.utilities.getOrThrow
+import org.junit.Test
+import org.web3j.crypto.Hash
+import kotlin.test.assertNotNull
+import com.r3.corda.interop.evm.common.trie.PatriciaTrie
+import org.web3j.rlp.RlpEncoder
+import org.web3j.rlp.RlpString
+import org.web3j.utils.Numeric
+import kotlin.test.assertEquals
+
+class ProofTests : TestNetSetup() {
+
+    private val eventSignature = "Transfer(address,address,uint256)"
+    private val signatureHash: String = Numeric.toHexString(Hash.sha3(eventSignature.toByteArray()))
+    private val zeroAddress = "0".repeat(64)
+
+    override fun onNetworkSetup() {
+    }
+
+    @Test
+    fun `can prove inclusion of event in a block`() {
+        val amount = 1.toBigInteger()
+
+        // create an ERC20 Transaction that will emit a Transfer event
+        val transactionReceipt: TransactionReceipt = alice.startFlow(
+            Erc20TransferFlow(goldTokenDeployAddress, bobAddress, amount)
+        ).getOrThrow()
+
+        // Retrieve the event from the receipt logs. The event will have references to the
+        // transaction that generated it, and the block that mined the transaction.
+        val transferEvent = transactionReceipt.logs?.single { log ->
+            // Topic0 = event signature
+            log.topics?.firstOrNull() == signatureHash &&
+            // Topic1 = address indexed `Transfer from`
+            log.topics?.getOrElse(1) { _ -> zeroAddress }?.takeLast(40) == Numeric.cleanHexPrefix(aliceAddress) &&
+            // Topic2 = address indexed `Transfer to`
+            log.topics?.getOrElse(2) { _ -> zeroAddress }?.takeLast(40) == Numeric.cleanHexPrefix(bobAddress) &&
+            // data = uint256 non-indexed `Transfer amount` - single non-indexed value = 32 bytes long string
+            log.data?.let { Numeric.toBigInt(it) } == amount
+        }
+        assertNotNull(transferEvent, "Expected Transfer event not found")
```

Comment:
Logical flow seems a bit strange to use the ?. operator above but fail the assertion here, could just add to the final ?: operator or test and then do the boolean checks in separate lines of code for readability

Reply:
I see your concern about readability and the use of the safe call operator ?.. Given the nature of the data we're dealing with, we've had to handle nullable types due to their origins in Java, and because the Ethereum data structures we work with mix legacy and new versions leaving even more uncertainty on whether a field will be null or the consequences of initialising it with a default value.

We are planning to review the nullability of these data structures in the future to address this issue. For the time being, I will consider your suggestion to perform null checks and boolean checks in separate lines of code to improve readability